specs/eventing/control-plane.md

```diff
@@ -0,0 +1,1120 @@
+pattern is chosen to make it clear that changing the `spec.channel` is not an
+atomic operation, as it may span multiple storage systems. Changes to
+`spec.subscriber`, `spec.reply`, `spec.delivery` and other fields SHOULD be
+permitted, as these could occur within a single storage system.
+
+When a Subscription becomes associated with a channel (either due to creating
+the Subscription or the channel), the Subscription MUST only set the `Ready`
+condition to `true` after the channel has been configured to send all future
+events to the Subscriptions `spec.subscriber`. The Channel MAY send some events
+to the Subscription before prior to the Subscription's `Ready` condition being
+set to `true`. When a Subscription is deleted, the Channel MAY send some
+additional events to the Subscription's `spec.subscriber` after the deletion.
+
+<!--
+TODO: channel-compatible CRDs (Channelable)
+-->
+
+### Addressable Resolution
+
+Both Trigger and Subscription have optional object references (`ref` in
+`spec.subscriber`, `spec.delivery.deadLetterSink`, and `spec.reply` for
+Subscription) which are expected to conform to the Addressable partial schema
+("duck type"). An object conforms with the Addressable partial schema if it
+contains a `status.address.url` field containing a URL which may be used to
+deliver CloudEvents over HTTP. As a special case, Kubernetes `v1` `Service`
+objects are considered to have a `status.address.url` of
+`http://<service-dns-name>/`.
+
+If both the `ref` field and the `uri` fields are set on a Destination, then the
+Destination's address is the `uri` interpreted relative to the resolved URI of
+the `ref` field. This may be used, for example, to refer to a specific URL off a
+referenced domain name, like so:
+
+```yaml
+subscriber:
+  ref:
+    apiVersion: v1
+    kind: Service
+    name: test
+  uri: "/update
+```
+
+This Destination would resolve to
+`http://test.<current-namespace-dns-prefix>/update`, as `/update` would be
+interpreted relative to the `test` Service's DNS name.
+
+If one of these object references points to an object which does not currently
+satisfy this partial schema (either because the `status.address.url` field is
+empty, or because the object does not have that field), then the Trigger or
+Subscription MUST indicate an error by setting the `Ready` condition to `false`,
+and SHOULD include an indication of the error in a condition reason or type.
+
+Both Broker and Channel MUST conform to the Addressable partial schema.
+
+## Event Routing
+
+Note that the event routing description below does not cover the actual
+mechanics of sending an event from one component to another; see
+[the data plane](./data-plane.md) contracts for details of the event transfer
+mechanisms.
+
+### Content Based Routing
+
+A Broker MUST publish a URL at `status.address.uri` when it is able to receive
+events. This URL MUST accept CloudEvents in both the
```

Comment:
`status.address.uri` -> `status.address.url`

Reply:
"I" and "l" look so alike!

Thank you!